README.md

```diff
@@ -1,53 +1,109 @@
-`.github/wiki` (or other specified directory) contents.
+</div>
 
-```bash
-git clone https://github.com/octocat/my-repo.wiki.git .github/wiki
-rm -rf .github/wiki/.git
-git add .github/wiki
-git commit -m "Add wiki to repo"
-```
+🌐 Works across repositories (with a [PAT]) \
+📚 Pretty interface for Markdown docs \
+⤴️ Lets you open PRs for wiki docs
 
 ## Usage
 
-Unfortunately, `${{ github.token }}` does not have access to updating wiki
-repositories, so it is necessary to create a new PAT.
+❗ Make sure you turn on the GitHub wiki feature in your repo's settings menu.
+You'll also need to _manually_ create a dummy page to initialize the git repo
+that powers the GitHub wiki. If you don't, when we push to `your-repo.wiki.git`,
+your workflow will fail because the wiki doesn't exist.
+
+Create a workflow file named something like `.github/workflows/wiki.yml` and add
+a job that uses this action to it!
+
+```yml
+name: Wiki
+on:
+  push:
+    branches: [main]
+    paths: [wiki/**, .github/workflows/wiki.yml]
+concurrency:
+  group: wiki
+  cancel-in-progress: true
+permissions:
+  content: write
+jobs:
+  wiki:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: spenserblack/actions-wiki@v0.1.1
+```
+
+⚠️ Make sure you enable the `content: write` permission! GitHub recently
+[changed the default permissions granted to jobs] for new repositories.
 
-Start by creating a classic token. If your repository is public, you only need the
-`public_repo` scope. Otherwise, you must grant the full `repo` scope. Then, add
-the token to your repository secrets so that it can be used in your workflow.
+☁️ If you're pushing to a wiki that's **not the current repository** you'll need
+to get a [GitHub PAT] to push to it.
 
-Simple example usage can be found in [this workflow file](./.github/workflows/update-wiki.yml).
+### Options
+
+<!--
+Prettier doesn't format inline HTML. See #8480 on GitHub Issues. [1] Instead,
+You'll need to either wrap this in an ```html code block to abuse the formatter
+and then remove the ```html wrapper, or paste it into the online Prettier
+playground [2] and copy the result.
+
+[1]: https://github.com/prettier/prettier/issues/8480
+[2]: https://prettier.io/playground/
+-->
+
+<dl>
+  <dt><code>repository</code></dt>
+  <dd>
+    <b>Default:</b> <code>github.com/${{ github.repository }}</code><br />
+    The repository housing the wiki
+  </dd>
+  <dt><code>token</code></dt>
+  <dd>
+    <b>Default:</b> <code>${{ github.token }}</code><br />
+    The token to use when cloning and pushing wiki changes
+  </dd>
+  <dt><code>wiki-directory</code></dt>
+  <dd>
+    <b>Default:</b> <code>.github/wiki</code><br />
+    The directory to use for your wiki contents
+  </dd>
+  <dt><code>commit-message</code></dt>
+  <dd>
+    <b>Default:</b> <code>Update wiki ${{ github.sha }}</code><br />
+    The message to use when committing to the wiki
+  </dd>
+</dl>
```

Comment:
>table vs `<dl>` vs headings vs code or whatever?

I'd prefer to avoid HTML TBH, to make the plain-text more readable. So I think I'd prefer a table here, like I did in [this action](https://github.com/spenserblack/actions-tag-to-release#inputs).

However, making a Markdown table well-spaced is a bit of a pain to do manually, so maybe it's time to add `prettier` to the toolchain?

Reply:
> However, making a Markdown table well-spaced is a bit of a pain to do manually, so maybe it's time to add `prettier` to the toolchain?

Even adding prettier won't do anything to help the rendered version. If your descriptions are good, they should ideally be like two sentences. One to describe what the option _is_, and one to describe what the option _does_ (is used for). Like:

```
repo
This is the slug of the GitHub repository, optionally
prefixed with a domain name. We will push the wiki
folder chosen by the path option to this repo's wiki
```

The point is that these descriptions _can get long-ish_ and they don't fit well in a table when rendered like this: (from your own example lol)

![image](https://user-images.githubusercontent.com/61068799/225368196-2a898951-5ed9-4d1b-807c-ddf7fac09bee.png)

like wtf is that multiline `${{ github.token }}` thing?

---

Point is there are these options that I've thought of: (see top PR original post for `<details>` sections examples of each)

- table
- in-yaml `<pre>` block
- literal list `<ul>`
- heading sections for each option
- `<dl>` list

Here's my rudimentary analysis of the options:

- table: ugly multiline madness since you can't control width of columns
- pre: kinda cheating, but seems to be a decent middle-ground of src vs render
- list: 2d option tuples don't really fit into a 1d list
- heading: works ok, just takes up too much vertical space with all the margins
- dl: looks pretty decent on render, hard to markup in source

To make the `<dl>` _slightly_ better in source, you could

```md
<dl><dt><code>repo</code></dt><dd>

You can use plain \
**Markdown** here since there's _2_ newlines.

</dd><dt><code>option-b</code></dt>...
```
but some might argue (even me) that ☝️ is worse

We should open a discussion about this 😜

Reply:
Linking the relevant discussion: https://github.com/spenserblack/actions-wiki/discussions/20